### PR TITLE
[WIP] Docs

### PR DESCRIPTION
**Pull request by codebytere**
Adds documentation around usage and local development of trop

/cc @MarshallOfSound

### CODE REVIEW

File: docs/local-setup.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    - **Nota Bene:** For development we recommend enabling everything and then removing uncessary permissions later.
+4. Download the private key as `private-key.pem` into the repository’s directory
+5. Start the app with APP_ID=1234 npm start where 1234 is your GitHub App’s ID
+5. Update your GitHub App’s Webhook URL to your [smee.io](https://smee.io/) URL.
+6. Run `$ npm start` to start the server.
+
+### Debugging
+
+1. Always run $ npm install and restart the server if package.json has changed.
+  - To turn on verbose logging, start server by running: $ LOG_LEVEL=trace npm start
```
Comment (ckerr) [inline docs/local-setup.md]: I don't understand why this is a sub-item. It seems unrelated to list item 1.

File: docs/usage.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
+## Setting Up `trop`
+
+Welcome! We're glad you want to try out `trop`.
+
+#### What Does `trop` Do?
+
+This bot is designed to automate the process of backporting to versioned release branches.
```
Comment (ckerr) [inline docs/usage.md]: omit unnecessary words: "is designed to"
omit unnecessary words: "the process of"

Trop automates backporting PRs to versioned release branches.

File: docs/usage.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+To install this app and have it work properly you will need:
+1. A project board with column names corresponding to your backport labels
+  - Ex. a column name of `target/2-0-0` will correspond to a label of `target/2-0-0`
```
Comment (ckerr) [inline docs/usage.md]: It's a simple thing, but very nice to have an example here :smile:

File: docs/usage.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+2. A `.github/config.yml` file with fields as specified in [this example config](.example.config)
```
Comment (ckerr) [inline docs/usage.md]: A `.github/config.yml` file. See [this example](.example.config)

File: docs/usage.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+The flow of this app is as follows:
```
Comment (ckerr) [inline docs/usage.md]: Using Trop:

File: docs/usage.md [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+1. Open a bugfix or feature pull request to `master`
```
Comment (ckerr) [inline docs/usage.md]: Feature request: this should be configurable, some projects may want to port from somewhere else

File: docs/usage.md [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+2. Tag the pull request with one of the backport labels (ex. `target/2-0-0`)
```
Comment (ckerr) [inline docs/usage.md]: Add a backport label to the pull request (ex. `target/2-0-0`)

File: docs/usage.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+3.`trop` will immediately place this pull request as a note into the corresponding column of the `watchedProject` as specified in your config file
```
Comment (ckerr) [inline docs/usage.md]: * s/immediately place/add/
* "as a note" -- not sure if this is relevant? If not, remove entirely
* s/as specified//

File: docs/usage.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+4. You or another co-contributor merges the pull request into `master`
```
Comment (ckerr) [inline docs/usage.md]: * s/another //
* s/merges/reviews and merges/ :smiling_imp:

File: docs/usage.md [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+5. Trop will automatically open pull requests containing `cherry-pick`s of the code into the backporting branches you specified in your labels (in this case, `2.0.0`).
+6. You or a co-contributor resolves conflicts if they exist and merges in the automatically created backport branches
```
Comment (ckerr) [inline docs/usage.md]: * s/conflicts if they exist/any conflicts/
* s/in the automatically created backport branches/trop's backports/

File: docs/usage.md [hunk 11]
```diff
@@ -0,0 +1,26 @@
...
+7. `trop` will move the cards it created for this backport into corresponding `merged/2-0-0` columns in the specified `watchedProject` board.
```
Comment (ckerr) [inline docs/usage.md]: into corresponding columns (ex. `merged/2-0-0`)

File: docs/usage.md [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+
+Congratulations! You've significantly reduced friction in your backporting process :tada:
```
Comment (ckerr) [inline docs/usage.md]: I hate everything about this sentence.

Better: "That's all there is to it! Congratulations, you did it! :tada:"

File: docs/local-setup.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+$ cd trop
+$ npm install
+```
+
+### Configuring the GitHub App
+
+To run your app in development, you will need to configure a GitHub App to deliver webhooks to your local machine.
+
+1. Go to [smee.io](https://smee.io/) and click **Start a new channel**.
+2. Set `WEBHOOK_PROXY_URL` in `.env` to the URL that you are redirected to.
```
[the remark this answers is not in the document]
Comment (codebytere) [reply docs/local-setup.md]: oh it comes by default in `create-probot-app` but im also just now realizing it's ignored, so the user should create one. I'll explain that further

File: docs/.example.config [hunk 14]
```diff
@@ -0,0 +1,6 @@
+watchedProject:
+  name: Radar
```
[the remark this answers is not in the document]
Comment (codebytere) [reply docs/.example.config]: yep, changed to `YOUR_PROJECT_NAME`

File: docs/local-setup.md [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+$ npm install
+```
+
+### Configuring the GitHub App
+
+To run your app in development, you will need to configure a GitHub App to deliver webhooks to your local machine.
+
+1. Go to [smee.io](https://smee.io/) and click **Start a new channel**.
+2. Create a `.env` file (example found [here](.example.env))
+2. Set `WEBHOOK_PROXY_URL` in `.env` to the URL that you are redirected to.
```
Comment (ckerr) [inline docs/local-setup.md]: 1. Duplicated numbers here and below (two #2, two #5)
2. I like how you phrase this below, "your `.env` file"

File: docs/.example.config [hunk 16]
```diff
@@ -0,0 +1,6 @@
+watchedProject:
+  name: YOUR_PROJECT_NAME
+
+authorizedUsers:
```
Comment (ckerr) [inline docs/.example.config]: :tada:
